packages/mdc-textfield/helper-text/README.md

```diff
@@ -0,0 +1,133 @@
+
+<ul class="icon-list">
+  <li class="icon-list-item icon-list-item--spec">
+    <a href="https://material.io/guidelines/components/text-fields.html#text-fields-layout">Material Design guidelines: Text Fields Layout</a>
+  </li>
+</ul>
+
+
+## Usage
+
+### Using helper text
+
+MDC Text Fields can include helper text that is useful for providing supplemental
+information to users, as well for validation messages (covered below).
+
+```html
+<div class="mdc-text-field">
+  <input type="text" id="username" class="mdc-text-field__input" aria-controls="username-helper-text">
+  <label for="username" class="mdc-text-field__label">Username</label>
+  <div class="mdc-text-field__bottom-line"></div>
+</div>
+<p id="username-helper-text" class="mdc-text-field-helper-text" aria-hidden="true">
+  This will be displayed on your public profile
+</p>
+```
+
+Helper text appears on input field focus and disappears on input field blur by default when using
+the Text Field JS component.
+
+#### Persistent helper text
+
+If you'd like the helper text to always be visible, add the `mdc-text-field-helper-text--persistent` modifier class to the element.
+
+```html
+<div class="mdc-text-field">
+  <input type="email" id="email" class="mdc-text-field__input">
+  <label for="email" class="mdc-text-field__label">Email address</label>
+  <div class="mdc-text-field__bottom-line"></div>
+</div>
+<p class="mdc-text-field-helper-text mdc-text-field-helper-text--persistent">
+  We will <em>never</em> share your email address with third parties
+</p>
+```
+
+#### Helper text and accessibility
+
+Note that in every example where the helper text is dependent on the state of the input element, we
+assign an id to the `mdc-text-field-helper-text` element and set that id to the value of the
+`aria-controls` attribute on the input element. We recommend doing this as well as it will help
+indicate to assistive devices that the display of the helper text is dependent on the interaction with
+the input element.
+
+When using our vanilla JS component, if it sees that the input element has an `aria-controls`
+attribute, it will look for an element with the id specified and treat it as the text field's help
+text element, taking care of adding/removing `aria-hidden` and other a11y attributes. This can also
+be done programmatically, which is described below.
+
+### Validation
+
+MDC TextField provides validity styling by using the `:invalid` and `:required` attributes provided
+by HTML5's form validation API.
+
+```html
+<div class="mdc-text-field">
+  <input type="password" id="pw" class="mdc-text-field__input" required minlength=8>
+  <label for="pw" class="mdc-text-field__label">Password</label>
+  <div class="mdc-text-field__bottom-line"></div>
+</div>
+```
+
+By default an input's validity is checked via `checkValidity()` on blur, and the styles are updated
+accordingly. Set the MDCTextField.valid variable to set the input's validity explicitly. MDC TextField
```

Comment:
Set the MDCTextField.valid field....not variable.

Reply:
Oops some of this section isn't actually related to helper text....moved it back to the parent README and kept only the parts relevant to helper text.